lib/ansible/modules/package.py

```diff
@@ -18,6 +18,7 @@
 short_description: Generic OS package manager
 description:
      - Installs, upgrade and removes packages using the underlying OS package manager.
+     - Provides only a subset of features provided by native package manager.
```

Comment:
This isn't an accurate statement.  The `package` module doesn't provide any features of it's own in reality; It's just a proxy.  We simply pass on the arguments to the underlying package module.  There is no filtering or otherwise done.

Reply:
> 
> This isn't an accurate statement. The `package` module doesn't provide any features of it's own in reality; It's just a proxy.

I understand and this is interesting so I can add that. I used the `provide` in the sense from the reader point of view (which is not perhaps a developer) I'm not sure it matters. Do you have any better wording ?

Reply:
I'd probably move it to `notes` and then also something like:

> This module acts as a proxy to the underlying package manager module. While all arguments will be passed to the underlying module, not all modules support the same arguments. This documentation only covers the minimum intersection of module arguments that all packaging modules support.

Combined with a statement that when dealing with multiple distributions, that is may be more useful to call the specific packaging module will often be more flexible, due to the reasons mentioned above.

Reply:
> This documentation only covers the minimum intersection of module arguments that all packaging modules support.

so as I understand it, this means the module could accept more options as long as all packaging system support them. Not sure I want to explain something which is not meant to be supported and encouraged.

Reply:
It's a fine line indeed.  But we often get issues that say "add support for X in package module" and it already supports it, but the documentation doesn't expose it.  Because of the lack of documenting how this module does what it does, people are always mislead into incorrect assumptions.

I'd rather be correct, and then use that correctness to explain the reasoning why you might want to just use the underlying module.

Regardless of whether we encourage something, we owe it to people to be correct.  We don't encourage anyone to use `lineinfile` but it still exists, and doesn't contain neutered documentation that would cause someone to not use it.